### PR TITLE
Freenas/11.1 stable ticket 40720

### PR DESCRIPTION
**Pull request by jhixson74**
DEPENDS: https://github.com/freenas/build/pull/96

### CODE REVIEW

File: src/freenas/etc/directoryservice/LDAP/ctl [hunk 1]
```diff
@@ -164,6 +198,11 @@ ldapctl_stop()
 		sssd_stop
 		ldapctl_cmd ${service} ix-sssd start
 	fi
+	if nslcd_running
+	then
+		nslcd_stop
+		ldapctl_cmd ${service} ix-nslcd start
```
Comment (b-a-t) [inline src/freenas/etc/directoryservice/LDAP/ctl]: I couldn't find `ix-nslcd` in the master, does it exist or it should be `nslcd` here?

Comment (jhixson74) [reply src/freenas/etc/directoryservice/LDAP/ctl]: Good question. I don't know! I'll look at the patch and see what I meant to do here.

File: src/middlewared/middlewared/etc_files/local/nslcd.conf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            return val
+
+        ldap = safe_call('datastore.query', 'directoryservice.LDAP')
+        if ldap and ldap[0]:
+            ldap = ldap[0]
+            capath = None
+            if ldap['ldap_certificate']:
+                cert = safe_call('certificateauthority.query', [('id', '=', ldap['ldap_certificate']['id'])], {'get': True})
+                if cert:
+                    capath = cert['cert_certificate_path']
```
Comment (b-a-t) [inline src/middlewared/middlewared/etc_files/local/nslcd.conf]: I have no idea what I'm talking about but in the next file there is an update:
```                if cert:
-                    capath = cert['cert_certificate_path']
+                    capath = cert['certificate_path']```
So one of them must be wrong, I guess.

File: src/middlewared/middlewared/etc_files/local/nslcd.conf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        else:
+            ldap = None
+
+        ldap_enabled = safe_call('notifier.common', 'system', 'ldap_enabled')
+
+        ldap_uri = "%s://%s" % ("ldaps" if ldap['ldap_ssl'] == "on" else "ldap", ldap['ldap_hostname'])
```
Comment (b-a-t) [inline src/middlewared/middlewared/etc_files/local/nslcd.conf]: Potentially, `ldap` can be set to `None` here, so querying `ldap['ldap_hostname']` will give us an exception here?

File: src/middlewared/middlewared/etc_files/local/nss_ldap.conf [hunk 4]
```diff
@@ -13,7 +13,7 @@
             if ldap['ldap_certificate']:
                 cert = safe_call('certificateauthority.query', [('id', '=', ldap['ldap_certificate']['id'])], {'get': True})
                 if cert:
-                    capath = cert['cert_certificate_path']
```
Comment (b-a-t) [inline src/middlewared/middlewared/etc_files/local/nss_ldap.conf]: This change vs. file above.

File: src/middlewared/middlewared/etc_files/local/nslcd.conf (hunk 2 above)
Comment (jhixson74) [reply src/middlewared/middlewared/etc_files/local/nslcd.conf]: So apparently cert_certificate_path is the correct key. This was changed at some point, which is why you see a bunch of cert_certificate_path to certificate_path changes. Now that I see that cert_certificate_path is the correct key,  I need to revisit all of these PR's where that key change is made. Good find.

**Pull request merged**